lib/internal/fs/streams.js

```diff
@@ -79,6 +112,30 @@ function close(stream, err, cb) {
   }
 }
 
+function importFd(stream, options) {
+  stream.fd = null;
+  if (options.fd) {
+    // When fd is a FileHandle we can listen for 'close' events
+    if (options.fd instanceof FileHandle) {
+      // FileHandle is not supported with custom fs operations
+      if (options.fs)
+        throw new ERR_METHOD_NOT_IMPLEMENTED('FileHandle with fs');
+      stream[kHandle] = options.fd;
+      stream.fd = options.fd.fd;
+      stream[kFs] = FileHandleOperations(stream[kHandle]);
+      stream[kHandle][kRef]();
+      options.fd.on('close', stream.close.bind(stream));
```

Comment:
`stream.close` is a `Function`, right?
```suggestion
      options.fd.on('close', FunctionPrototypeBind(stream.close, stream));
```